### ISSUE TITLE
Docker 1.7.0-rc1 did not compile on ARM

### ISSUE THREAD

**Issue by DieterReuter** [docker/docker#13534]
Trying to compile Docker-1.7.0-rc1 on an ARMv7l machine (Scaleway Cloud), I get the following error:
```
---> Making bundle: dynbinary (in bundles/1.7.0-rc1/dynbinary)
# github.com/vishvananda/netns
vendor/src/github.com/vishvananda/netns/netns_linux_arm.go:4: SYS_SETNS redeclared in this block
	previous declaration at vendor/src/github.com/vishvananda/netns/netns_linux_amd.go:4
```
Seems to be that the dependency `github.com/vishvananda/netns` breaks it.

*Workaround:*
Removing `./vendor/src/github.com/vishvananda/netns/netns_linux_amd.go` gets me around this error. Now it's compiling successfully.

**Comment by hqhq** [comment]
Hi, you can add `// +build linux,amd` to `vendor/src/github.com/vishvananda/netns/netns_linux_amd.go` to work around this, the latest netns has fixed this.

**Comment by DieterReuter** [comment]
@hqhq, thanks for this info. Now I know, we're just waiting to include support for arm64 to fix this issue as well! Good to know.

**Comment by thaJeztah** [comment]
I'm going to close this issue, because this looks to be an issue in a vendored package, not in docker. I suggest to open an issue in the repo for that package.

Feel free to continue the discussion after closing

**Comment by runcom** [comment]
I think we should update `netns` dependency if we care about building on arm

**Comment by thaJeztah** [comment]
Oops, I overlooked this;

> the latest netns has fixed this.

@hqhq @DieterReuter feel free to create a PR to update the netns version if they have fixed this already, I'll re-open

**Comment by mavenugo** [comment]
@hqhq @thaJeztah @DieterReuter This came in via libnetwork upgrade. And there was a recent discussion around this very same topic. @dave-tucker & @hqhq were involved working on the `netns` library. They might have more info to share. In short, this update must happen in libnetwork and must get into the 1.7.0 bump branch through vendor-in process.

**Comment by thaJeztah** [comment]
@mavenugo ah! Thanks for the reminder, I though I've seen a similar discussion, but couldn't find it, because I was looking in the wrong repo.

Here's what I'll do; I'll keep this issue open here, if one of you can update the vendored package in libnetwork, then bump the version here; please mention a `closes #13534`

If this is needed in the 1.7 release; we're in feature-freeze, but perhaps it could be cherry-picked if needed.

**Comment by mavenugo** [comment]
@thaJeztah We are planning on a vendor-in of libnetwork shortly in the 1.7.0 bump branch shortly with selected bug-fixes that were marked for the 1.7.0 release. We can accomodate this netns library upgrade once @dave-tucker and @hqhq solve the issues around this.

**Comment by thaJeztah** [comment]
@mavenugo thanks for the update, please remind me to close this after that, or feel free to close now if you think that's appropriate :smile:

**Comment by dave-tucker** [comment]
@thaJeztah @mavenugo there is no issue with armv6/armv7 in the updated `netns`. The patch I'm working on with @hqhq it to address armv8 and I'm currently blocked on getting some hardware (or working qemu) to recreate the issue

**Comment by thaJeztah** [comment]
@dave-tucker Perhaps @DieterReuter can help providing access to the hardware? I might still have my scaleway test account active, but it may be worth a try contacting them directly (as it's in their interest too)

**Comment by thaJeztah** [comment]
Oh, realize I might have misread your comment; Scaleway is on Armv7 I think?

**Comment by dave-tucker** [comment]
Yep :) armv7 and armv6 should be fine if we just bump the `netns` dependency. The issue Madhu mentioned was related to armv8 and that's where I'm currently blocked ;)

**Comment by DieterReuter** [comment]
@dave-tucker, you're absolute right, `armv6l` and `armv7l` is just fine with new version of `netns`. 
@thaJeztah, ScaleWay is `armv7l`.

**Comment by DieterReuter** [comment]
I just tested again with 1.7.0-rc4, this issue is fixed now.

**Comment by thaJeztah** [comment]
Thanks for reporting back, @DieterReuter! And good to hear :)

**Comment by DieterReuter** [comment]
Issue fixed by #13784.
